### PR TITLE
feat(docs): add a basic eslint doc

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

ChatGPT and popular guides were all incorrect or outdated. This guide isn't great but at least it helps spotlight that linting in Expo is basic and could stand to be improved.

### CODE REVIEW

File: docs/pages/guides/using-eslint.mdx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+const config = getDefaultConfig(
+  /* @info This will no longer assert linting errors */ __dirname /* @end */
+);
+
+module.exports = config;
+```
+
+## Troubleshooting
+
+### ESLint is updating in VS Code
```
Comment (EvanBacon) [inline docs/pages/guides/using-eslint.mdx:105]:
```suggestion
### ESLint is not updating in VS Code
```

File: docs/pages/guides/using-eslint.mdx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+}
+```
+
+You may want to replace the `.` with specific directories or files to lint. For example, if you use Expo Router, you can use `eslint app` to lint only your routes.
+
+</Step>
+
+## Usage
+
+You can lint your code manually from the command-line with with the script:
```
Comment (amandeepmittal) [inline docs/pages/guides/using-eslint.mdx:70]:
```suggestion
You can lint your code manually from the command-line with the script:
```

File: docs/pages/guides/using-eslint.mdx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+```json package.json
+{
+  "scripts": {
+    "lint": "eslint ."
+  }
+}
+```
+
+You may want to replace the `.` with specific directories or files to lint. For example, if you use Expo Router, you can use `eslint app` to lint only your routes.
```
Comment (amandeepmittal) [inline docs/pages/guides/using-eslint.mdx:64]:
```suggestion
You can replace the `.` with specific directories or files to lint. For example, if you use Expo Router, you can use the `eslint app` command to lint only your routes inside the **app** directory.
```

File: docs/pages/guides/using-eslint.mdx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        <Terminal cmd={['$ yarn lint']} />
+    </Tab>
+
+</Tabs>
+
+If you're using VS Code, you can install the [ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) to lint your code as you type (recommended).
+
+## Envrionment configuration
+
+ESLint is generally configured for a single environment, the source code in an Expo app, while written in JavaScript, is run in multiple different environments. For example, the **app.config.js**, **metro.config.js**, **babel.config.js**, and **app/+html.tsx** files are run in Node.js envrionments, this means they have access to the global `__dirname` variable, and can use Node.js modules like `path`. Standard Expo project files like **app/index.js** can be run in Hermes, Node.js, or the web browser.
```
Comment (amandeepmittal) [inline docs/pages/guides/using-eslint.mdx:87]:
```suggestion
ESLint is generally configured for a single environment. However, the source code is written in JavaScript in an Expo app runs in multiple different environments. For example, the **app.config.js**, **metro.config.js**, **babel.config.js**, and **app/+html.tsx** files are run in a Node.js environment. It means they have access to the global `__dirname` variable and can use Node.js modules such as `path`. Standard Expo project files like **app/index.js** can be run in Hermes, Node.js, or the web browser.
```

- The first sentence has too many pauses, so created separation by splitting it into two sentences.
- Fixed grammatical errors

File: docs/pages/guides/using-eslint.mdx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+You can add the `eslint-env` comment directive to the top of a file to tell ESLint which environment the file is run in. For example, to tell ESLint that a file is run in Node.js, add the following comment to the top of the file:
```
Comment (amandeepmittal) [inline docs/pages/guides/using-eslint.mdx:89]:
```suggestion
You can add the `eslint-env` comment directive to the top of a file to tell ESLint which environment the file is running in. For example, to tell ESLint that a file is run in Node.js, add the following comment to the top of the file:
```

**Review by amandeepmittal — APPROVED** [review]
Much needed 👏 

Thank you for adding this guide. Left some suggestions to improve verbiage but other wise lgtm 👍

**Pull request merged**